infrastructure-playbooks/shrink-mgr.yml

```diff
@@ -0,0 +1,128 @@
+---
+# This playbook shrinks the Ceph manager from your cluster
+#
+# Use it like this:
+# ansible-playbook shrink-mgr.yml -e mgr_to_kill=ceph-mgr1
+#     Prompts for confirmation to shrink, defaults to no and
+#     doesn't shrink the cluster and yes shrinks the cluster.
+#
+# ansible-playbook -e ireallymeanit=yes|no shrink-mgr.yml
+#     Overrides the prompt using -e option. Can be used in
+#     automation scripts to avoid interactive prompt.
+
+
+- name: gather facts and check the init system
+  hosts:
+    - "{{ mon_group_name | default('mons') }}"
+    - "{{ mgr_group_name | default('mgrs') }}"
+  become: true
+  tasks:
+    - debug:
+        msg: gather facts on all Ceph hosts for following reference
+
+- name: confirm if user really meant to remove manager from the ceph cluster
+  hosts: localhost
+  become: true
+  vars_prompt:
+    - name: ireallymeanit
+      prompt: Are you sure you want to shrink the cluster?
+      default: 'no'
+      private: no
+  pre_tasks:
+    - import_role:
+        name: ceph-defaults
+
+    - import_role:
+        name: ceph-facts
+
+    - name: exit playbook, if only one manager is present in cluster
+      fail:
+        msg: "You are about to shrink the only manager present in the cluster."
+      when: groups[mgr_group_name] | length | int < 2
+
+    - name: exit playbook, if no manager was given
+      fail:
+        msg: "mgr_to_kill must be declared
+              Exiting shrink-cluster playbook, no manager was removed.
+              On the command line when invoking the playbook, you can use
+              -e mgr_to_kill=ceph-mgr01 argument. You can only remove a single
+              manager each time the playbook runs."
+      when: mgr_to_kill is not defined
+
+    - name: exit playbook, if the manager is not part of the inventory
+      fail:
+        msg: "It seems that the host given is not part of your inventory,
+              please make sure it is."
+      when: mgr_to_kill not in groups[mgr_group_name]
```

Comment:
ditto

Reply:
Working on this now. I am trying to create a list of all MGRs in cluster.

Reply:
Updated the condtion to make sure that mgr_to_kill is either of active or standby MGRs.